### DISCUSSION TITLE
How long does it take to interpret model (SHAP)?

### DISCUSSION THREAD

**Discussion by augustaklug** [pycaret/pycaret#1202] [Q&A] [answered]
I'm creating a random forest regression model on a Kaggle notebook. The modeling data has 17 columns and 133934 rows. How long does it usually take to interpret the model (`interpret_model(rf)`)? My notebook has been running for about 2 hours, and still has no output. What can I do to speed up the process?

**Comment by Yard1** [comment, marked as answer]
SHAP is a resource intensive process. I'd recommend taking a 10-20% sample from the data and running SHAP on that. The results should be close to what you'd get with the full dataset.

**Reply by augustaklug** [reply]
How can I change the sample parameter?

**Reply by Yard1** [reply]
It's not a feature in PyCaret - just sample the data first before passing it to `setup()`.

**Reply by eddy-geek** [reply]
I'd say using something like this would do the trick (not sure):

```python
N = 100
interpret_model(best_model, X_new_sample=X[:N], y_new_sample=y[:N])
```
